qiskit_experiments/library/characterization/fine_amplitude.py

```diff
@@ -131,13 +135,20 @@ def _pre_circuit(self) -> QuantumCircuit:
         This method can be overridden by subclasses e.g. to calibrate gates on
         transitions other than the 0 <-> 1 transition.
         """
-        circuit = QuantumCircuit(1)
+        circuit = QuantumCircuit(1, 1)
 
         if self.experiment_options.add_sx:
             circuit.sx(0)
 
         return circuit
 
+    def _add_measure(self, circuit: QuantumCircuit):
+        """Add the measurements to the quantum circuit.
+
+        Sub-classes may override this function.
+        """
+        circuit.measure_all()
```

Comment:
If you want to allow subclasses overriding how many qubits are measured etc, this function could be changed to have signature `_measure_circuit(self) -> QuantumCircuit` in the base class with default being:
```python
circ = QuantumCircuit(self.num_qubits)
circ.measure_all()
return circ
```
and then in the cirucits function you could do something like.

```python

# Get meas circuit to allocate number of required clbits
meas_circ = self._measure_circ()

# Build cal circs
cal_circ = QuantumCircuit(self.num_qubits, meas_circ.num_clbits)

# make the cal circs ...

for repetition in repetitions:
     # ...
    cal_circ.append(meas_circ, range(meas_circ.num_qubits), range(meas_circ.num_clbits))
```

Reply:
Thanks for the suggestion. See here: [c5d7653](https://github.com/Qiskit/qiskit-experiments/pull/655/commits/c5d76532db2392785cca70c6972b72ff5224a77f) this is much more consistent.